cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -129,26 +131,24 @@ type containerProcess struct {
 }
 
 // preStart prepares for starting a container process and handling its exit.
-// The container being started should be passed in as c when starting the
-// container init process for an already-created container. c should be nil when
-// creating a container or when starting an exec.
+// The container being started should be passed in as c when starting the container
+// init process for an already-created container. c should be nil when creating a
+// container or when starting an exec.
 //
 // The returned handleStarted closure records that the process has started so
 // that its exit can be handled efficiently. If the process has already exited,
-// it handles the exit immediately. handleStarted should be called after the
-// event announcing the start of the process has been published.
-// Note that handleStarted needs to be aware of whether s.mu is already held
-// when it is called. If s.mu has been held, we don't need to lock it when
-// calling handleProcessExit.
+// it handles the exit immediately. In addition, if the process is an exec and
+// it's container's init process has already exited, that exit is also processed.
+// handleStarted should be called after the event announcing the start of the
+// process has been published. Note that handleStarted needs to be aware of
+// whether s.mu is already held when it is called. If s.mu has been held, we
+// don't need to lock it when calling handleProcessExit.
```

Comment:
Not any more. The returned closure no longer has a `muLocked` parameter.
```suggestion
// process has been published. Note that s.lifecycleMu must not be held when
// calling handleStarted.
```